shared/settings/settings.type.ts

```diff
@@ -0,0 +1,318 @@
+export type AllSettings = StaticSettings & DynamicSettings;
+
+export type StaticSettings = {
```

Comment:
We could add general descriptions for these types:

- `StaticSettings`: These settings can only be set by editing the `staticSettings.ts` file and they cannot, thus, change after initialisation.
- `DynamicSettings`: These settings can be set either by editing the `dynamicSettings.ts` file or overwritten with settings loaded by the `DataProvider`.

Also, the `AppSettings` type should be updated to be a combination of these two.

Could you also check:

- Is there a way to make the static settings readonly so they cannot be overwritten by data loaded by accident? If not, just typing the incoming data as `DynamicSettings` ought to be pretty robust.
- Is there a use case and an easy way to copy the current settings files as defaults and then add another pair of files where the contents are `Partial`s of the settings and are gitignored (with example files provided)? The cascade would thus be:

1. Defaults (ts-files): `Static/DynamicSettings`
2. Local overrides (ts-files based on example files, optional, gitignored): `Partial<Static/DynamicSettings>`
3. DataProvider-provided overrides (fetched, optional): `Partial<DynamicSettings>`

Reply:
Note that there might be a discrepancy with this system when adding dynamic settings if their defaults are based on the defaul local files. Maybe use the merged defaults + local overrides to set their default values?

Reply:
Let's add an empty local override file for now.

Reply:
I have worked on making settings in local override files and Strapi deeply `Partial`. One problem I encountered is that if `survey` hasn't been set in `defaultDynamicSettings.ts`, user could have following local override file:
```ts
// dynamicSettings.ts
export const dynamicSettings: DeepPartial<DynamicSettings> = {
    survey: {
        linkTemplate: "example"
    }
};
```
If this would be merged with default setting (with some generic deep merge function), resulting settings would have `survey` without required property `showIn`. So we would need to have some kind of more specialized and complex merge function.

It feels like these settings are getting pretty complicated. Is there really use case for these local orverride files? Wouldn't it work if we would have one file for dynamic settings and one file for static settings. Those files would be similar to default settings files in this branch, but user could modify those if needed. In Strapi, dynamic settings would be required and optional in the same way as in the settings type. Contents of settings file would be loaded into Strapi so user would see them and could modify even a single setting in there. Would there be any problems with this kind of approach?